### ISSUE TITLE
Print location of system SOCI in the build log

### ISSUE THREAD

**Issue by lazariv** [rstudio/rstudio#14426]
### System details

    RStudio Edition : Server
    RStudio Version : v2023.12.1+402
    OS Version      : RHEL 8.7
    R Version       : 4.3.3
    SOCI            : 4.0.3

### Steps to reproduce the problem
Build RStudio from source with system SOCI 

### Describe the problem in detail
When building latest RStudio Server from source the system-site SOCI libraries are found, but its location does not appear in the build-log, since `SOCI_LIBRARY_DIR` is not set in this case.

https://github.com/rstudio/rstudio/blob/83ef05e3eb6916b98780203fd3ff81da314f43cd/src/cpp/CMakeLists.txt#L541 
The output of the above source code line is:
```
SOCI libraries found under 
```

### Describe the behavior you expected
I expect `SOCI_LIBRARY_DIR` variable is set (in the case of `UNIX` and `RSTUDIO_USE_SYSTEM_SOCI` are true) and printed.

Alternatively the variables `SOCI_CORE_LIB` and `SOCI_SQLITE_LIB` are printed in the log. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@lazariv Thank you for reporting this! Were you able to build RStudio Server from source in previous releases?

**Comment by lazariv** [comment]
> @lazariv Thank you for reporting this! Were you able to build RStudio Server from source in previous releases?

Yes. At the very beginning RStudio searched for system SOCI only in the default path: `set(SOCI_LIBRARY_DIR "/usr/lib")`

This has been fixed with https://github.com/rstudio/rstudio/pull/11691 . Now only the information about the found SOCI libraries is missing.

**Comment by gtritchie** [comment]
Closing, nothing to validate here (other than ensuring the build isn't broken, which will happen automatically).
